pyop2/fusion.py

```diff
@@ -267,17 +267,31 @@ def __init__(self, kernel, it_space, *args, **kwargs):
             return
         self._all_args = kwargs.pop('all_args')
         self._executor = kwargs.pop('executor')
+        self._it_space = it_space
         super(JITModule, self).__init__(kernel, it_space, *args, **kwargs)
 
-    def compile(self, argtypes=None, restype=None):
-        if hasattr(self, '_fun'):
-            # It should not be possible to pull a jit module out of
-            # the cache /with/ arguments
-            if hasattr(self, '_args'):
-                raise RuntimeError("JITModule is holding onto args, memory leak!")
-            self._fun.argtypes = argtypes
-            self._fun.restype = restype
-            return self._fun
+    def set_argtypes(self, iterset, *args):
```

Comment:
This method doesn't really set anything, could be called `build_argtypes` rather.